lib/msf/core/payload/windows/reverse_winhttp.rb

```diff
@@ -117,9 +123,22 @@ def asm_reverse_winhttp(opts={})
     retry_count       = [opts[:retry_count].to_i, 1].max
     verify_ssl        = nil
     encoded_cert_hash = nil
-    encoded_url       = asm_generate_wchar_array(opts[:url])
+    encoded_uri       = asm_generate_wchar_array(opts[:uri])
     encoded_host      = asm_generate_wchar_array(opts[:host])
 
```

Comment:
is there any good way for winhttp/https to share this code? it seems like a lot to duplicate between them.